ksqldb-engine/src/main/java/io/confluent/ksql/planner/plan/JoinNode.java

```diff
@@ -254,16 +255,17 @@ private void ensureMatchingPartitionCounts(final KafkaTopicClient kafkaTopicClie
             + "number of partitions match.");
   }
 
-  private FormatInfo getDefaultSourceKeyFormat() {
+  private KeyFormat getDefaultSourceKeyFormat() {
     return Stream.of(left, right)
         .flatMap(PlanNode::getSourceNodes)
         .map(DataSourceNode::getDataSource)
         .map(DataSource::getKsqlTopic)
         .map(KsqlTopic::getKeyFormat)
-        .map(KeyFormat::getFormatInfo)
-        .filter(format -> !format.getFormat().equals(NoneFormat.NAME))
+        .filter(format -> !format.getFormatInfo().getFormat().equals(NoneFormat.NAME))
         .findFirst()
-        .orElse(FormatInfo.of(defaultKeyFormat));
+        // if none exist, assume non-Windowed since that would mean that both sources
+        // were of NONE format, which doesn't support windowed operations
+        .orElse(KeyFormat.nonWindowed(FormatInfo.of(defaultKeyFormat), SerdeFeatures.of()));
```

Comment:
Upon first reading I was confused why this wasn't being sanitized with `SerdeFeaturesFactory#sanitizeKeyFeatures()`. Looks like the reason is all the sanitization happens in SchemaKStream and SchemaKTable instead. Might be worth adding that into the javadocs for the method?